### PR TITLE
[TC-230] Feat/tc 230 profile

### PR DESCRIPTION
**Pull request by fw-noel**
[TC-230](https://bcdevex.atlassian.net/browse/TC-230)

Some todos:
- Figure out home location, is that data we will have?
- Icons for locations
- Edit info, keep or no?
- Show / Hide last row,  need or no?

Objective:
Screenshot 2024-02-02 at 1 47 56 AM

[TC-230]: https://bcdevex.atlassian.net/browse/TC-230?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: frontend/src/pages/profile/MemberDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import { useState } from 'react';
+import { Accordion, AccordionHeader, AccordionBody } from '@material-tailwind/react';
+import MemberDetailsSection from './MemberDetailsSection';
+import type { Personnel } from '../dashboard';
+import dayjs from 'dayjs';
+
+function ArrowIcon({ id, open }: { id: number; open: number }) {
```
Comment (chelsea-EYDS) [inline frontend/src/pages/profile/MemberDetails.tsx:7]: Split into Icons file in the UI directory?

File: frontend/src/pages/profile/MemberDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    </svg>
+  );
+}
+
+const MemberDetails = ({ personnel }: { personnel: Personnel }) => {
+  const [open, setOpen] = useState(1);
+
+  const handleOpen = (value: any) => setOpen(open === value ? 0 : value);
+
+  const generalInformation = [
```
Comment (chelsea-EYDS) [inline frontend/src/pages/profile/MemberDetails.tsx:31]: Split into const?

File: frontend/src/pages/profile/Profile.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    </svg>
+  );
+}
+
+const Profile = () => {
+  const { personnelId } = useParams() as { personnelId: string };
+  const { personnel } = usePersonnel({ personnelId });
+
+  return (
+    <div className="h-screen pt-12 pb-24 bg-grayBackground">
```
Comment (chelsea-EYDS) [inline frontend/src/pages/profile/Profile.tsx:44]: I think we need h-full here

File: frontend/src/pages/profile/MemberDetailsSection.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+const Detail = ({ title, content }: { title: string; content: string }) => {
```
Comment (chelsea-EYDS) [inline frontend/src/pages/profile/MemberDetailsSection.tsx:1]: Split into separate file?

File: frontend/src/pages/profile/MemberDetailsSection.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  return (
+    <div className="py-2">
+      <h5>{title}</h5>
+      <h4>{content}</h4>
```
Comment (chelsea-EYDS) [inline frontend/src/pages/profile/MemberDetailsSection.tsx:5]: I think we need <p> instead of <h4> here.

Comment (chelsea-EYDS) [reply frontend/src/pages/profile/MemberDetailsSection.tsx:5]: haha, sorry < p > instead of < h4 >

**Review by chelsea-EYDS — COMMENTED** [review]
Just some UI nits, otherwise looks great!!

File: frontend/src/pages/profile/Profile.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                <span className="pl-4 underline decoration-solid">
+                  Personnel (Dashboard)
+                </span>
+              </div>
+            </a>
+            <span className="font-bold">
+              {personnel.firstName} {personnel.lastName}
+            </span>
+          </Breadcrumbs>
+          <div className="pt-12">
```
Comment (chelsea-EYDS) [inline frontend/src/pages/profile/Profile.tsx:60]: Would you be open to splitting this into "Profile Header" as it's own component? (So this page would basically be a container for three components: Breadcrumbs, Profile Header, Member Details? (or, Profile Details?))

File: frontend/src/pages/profile/MemberDetails.tsx (hunk 1 above)
Comment (fw-noel) [reply frontend/src/pages/profile/MemberDetails.tsx:7]: Using hero icons now

File: frontend/src/pages/profile/MemberDetails.tsx (hunk 2 above)
Comment (fw-noel) [reply frontend/src/pages/profile/MemberDetails.tsx:31]: What do you mean?

Comment (chelsea-EYDS) [reply frontend/src/pages/profile/MemberDetails.tsx:13]: Oh sorry, I mean like another file

Comment (chelsea-EYDS) [reply frontend/src/pages/profile/MemberDetails.tsx:13]: and import

File: frontend/src/pages/profile/MemberDetailsSection.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  numColumns: number;
+  title: string;
+  columns: { title: string; content: string }[];
+}) => {
+  return (
+    <div className="py-5">
+      <h4 className="font-bold text-info">{title}</h4>
+      <div className={`grid grid-cols-${numColumns} pt-4`}>
+        {columns.map((column, i) => (
+          <div key={i}>
```
Comment (chelsea-EYDS) [inline frontend/src/pages/profile/MemberDetailsSection.tsx:17]: warning for using index as key . Not breaking, but maybe we can add the value + i.toString or something? I really don't know what the best practice is here.

**Review by chelsea-EYDS — DISMISSED** [review]
Looks great!

**Review by chelsea-EYDS — APPROVED** [review]

**Pull request merged**